engine/src/test/scala/cromwell/webservice/CromwellApiServiceSpec.scala

```diff
@@ -1,133 +1,55 @@
-
-  val file = WdlFile("/tmp/ps.stdout.tmp")
-  val fileHash = file.computeHash
 
   def receive = {
-//    case SubmitWorkflow(sources) =>
-//      val id = MockWorkflowManagerActor.submittedWorkflowId
-//      /*
-//        id will always succeed, but WorkflowDescriptor might not. If all of this works we
-//        want to pass Future[WorkflowId] but otherwise we want to pass the error. Try the
-//        WorkflowDescriptor - if it succeeds hand the id back in a future, otherwise the error
-//        from WorkflowDescriptor's validation
-//       */
-//      val message = Try(materializeWorkflowDescriptorFromSources(id = id, workflowSources = sources)) match {
-//        case Success(w) => WorkflowManagerSubmitSuccess(w.id)
-//        case Failure(e) => WorkflowManagerSubmitFailure(e)
-//      }
-//      sender ! message
-    case WorkflowStatus(id) =>
-      val message = id match {
-        case MockWorkflowManagerActor.runningWorkflowId => WorkflowManagerStatusSuccess(id, WorkflowRunning)
-        case MockWorkflowManagerActor.abortedWorkflowId => WorkflowManagerStatusSuccess(id, WorkflowAborted)
-        //case _ => WorkflowManagerStatusFailure(id, new WorkflowNotFoundException("Cromwell knows not this workflow"))
-      }
+    case SubmitWorkflowCommand(source) =>
+      val id = MockWorkflowManagerActor.submittedWorkflowId
+      val message = WorkflowManagerSubmitSuccess(id)
       sender ! message
-//    case WorkflowAbort(id) =>
-//      val message = id match {
-//        case MockWorkflowManagerActor.runningWorkflowId => WorkflowManagerAbortSuccess(id)
-//        case MockWorkflowManagerActor.abortedWorkflowId =>
-//          WorkflowManagerAbortFailure(id, new IllegalStateException(s"Workflow ID '$id' is in terminal state 'Aborted' and cannot be aborted."))
-//        case x => WorkflowManagerAbortFailure(id, new WorkflowNotFoundException(s"Workflow '$x' not found."))
-//      }
-//      sender ! message
-    case WorkflowOutputs(id) =>
+    case AbortWorkflowCommand(id) =>
       val message = id match {
-        case MockWorkflowManagerActor.submittedWorkflowId =>
-          WorkflowManagerWorkflowOutputsSuccess(id, Map(
-            "three_step.cgrep.count" -> JobOutput(int8, Option(int8hash)),
-            "three_step.ps.procs" -> JobOutput(file, Option(fileHash)),
-            "three_step.wc.count" -> JobOutput(int8, Option(int8hash))))
-        //case w => WorkflowManagerWorkflowOutputsFailure(id, new WorkflowNotFoundException(s"Workflow '$w' not found"))
+        case MockWorkflowManagerActor.runningWorkflowId => WorkflowManagerAbortSuccess(id)
+        case MockWorkflowManagerActor.abortedWorkflowId =>
+          WorkflowManagerAbortFailure(id, new IllegalStateException(s"Workflow ID '$id' is in terminal state 'Aborted' and cannot be aborted."))
+        case x =>
+          WorkflowManagerAbortFailure(id, new Exception(s"Couldn't abort $id because no workflow with that ID is in progress"))
```

Comment:
ToL: Is there a value in all this logic? Are we in danger of making tests which just test that we've mocked out the WorkflowManagerActor correctly?

I'd rather have different flavours of `MockWorkflowManagerActor` which, for example, always return a `WorkflowManagerAbortFailure`, or which always return `WorkflowManagerAbortSuccess` so we can reliably test what happens when we receive the various messages